expr.go

```diff
@@ -641,27 +641,34 @@ func evalExpr(e *expr, from, until int32, values map[metricRequest][]*metricData
 			return nil
 		}
 
+		re := regexp.MustCompile("^([a-z]+)([AB][a-z]+)$")
+		submatch := re.FindStringSubmatch(e.target)
+		var compute func([]float64, []bool) float64
+		switch submatch[1] {
+		case "average":
+			compute = avgValue
+		case "current":
+			compute = currentValue
+		case "maximum":
+			compute = maxValue
+		case "minimum":
+			compute = minValue
+		}
 		var results []*metricData
 		for _, a := range args {
-			switch e.target {
-			case "averageAbove":
-				if avgValue(a.Values, a.IsAbsent) >= float64(n) {
+			value := compute(a.Values, a.IsAbsent)
+			switch submatch[2] {
+			case "Above":
+				if value >= float64(n) {
 					results = append(results, a)
 				}
-			case "averageBelow":
-				if avgValue(a.Values, a.IsAbsent) <= float64(n) {
-					results = append(results, a)
-				}
-			case "currentAbove":
-				if currentValue(a.Values, a.IsAbsent) >= float64(n) {
-					results = append(results, a)
-				}
-			case "currentBelow":
-				if currentValue(a.Values, a.IsAbsent) <= float64(n) {
+			case "Below":
+				if value <= float64(n) {
```

Comment:
I the last "push -f" I maximumAbove and minimumAbove are exclusive as in the original Graphite. All the 8 methods are now consistent with the python version.